### PR TITLE
python311Packages.flask-sock: init at 0.7.0

### PR DESCRIPTION
**Pull request by fmhoeger**
## Description of changes

WebSocket support for Flask

https://github.com/miguelgrinberg/flask-sock

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Lurkki14** [comment]
You need to squash the commits to files `pkgs/development/python-modules/flask-sock/default.nix` and `pkgs/top-level/python-packages.nix` into one commit

**Comment by Lurkki14** [comment]
The commit message adding yourself to maintainers should be `maintainers: add fmhoeger`

**Comment by fmhoeger** [comment]
@Lurkki14 Thanks for the clear instructions!
Reworded and squashed my previous commits into now just two commits.

File: pkgs/development/python-modules/flask-sock/default.nix [hunk 1]
```diff
@@ -0,0 +1,53 @@
+{ lib
+, buildPythonPackage
+, pytestCheckHook
+, python-socketio
+, pythonOlder
+, coverage
```
Comment (Lurkki14) [inline pkgs/development/python-modules/flask-sock/default.nix:6]:
```suggestion
, pythonOlder
```
These are unused

**Review by Lurkki14 — CHANGES_REQUESTED** [review]

**Review by Lurkki14 — APPROVED** [review]

**Comment by fmhoeger** [comment]
Thanks for accompanying me on my first PR to nixpkgs!
I wonder if there is anything left to do for me?

**Comment by emilytrau** [comment]
Result of `nixpkgs-review pr 268194` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python310Packages.flask-sock</li>
    <li>python310Packages.flask-sock.dist</li>
    <li>python311Packages.flask-sock</li>
    <li>python311Packages.flask-sock.dist</li>
  </ul>
</details>

**Review by emilytrau — APPROVED** [review]
Welcome to nixpkgs!

**Pull request merged**